src/test/java/roboy/dialog/StateFactoryTest.java

```diff
@@ -28,7 +30,8 @@ public void factoryCreatesCorrectStateObjects() {
         classes.add(ToyIntroState.class);
 
 
-        StateParameters parms = new StateParameters(new DialogStateMachine());
+        InferenceEngine inference = new Inference();
+        StateParameters parms = new StateParameters(new DialogStateMachine(inference), inference);
```

Comment:
actually, StateParameters can get InferenceEngine directy from the DialogStateMachine
there is no need to explicitly pass InferenceEngine to the StateParameters as a parameter

Reply:
Refactored